CMakeModules/Detection.cmake

```diff
@@ -1,26 +1,26 @@
 set(BUILD_COMPILER "unknown")
```

Comment:
If we're renaming BUILD_PLATFORM and BUILD_ARCHITECTURE, let's also rename BUILD_COMPILER